inverse/beamformer_sam.m

```diff
@@ -123,136 +156,164 @@
-        [U,S] = eig(Y2,Y1);
-
+        [U,S] = eig(Y1,Y2);
+        
       case 'robert'
-        [U,S] = svd(real(pinv(L' * inv_cov * L)));
-
-      case 'stephen'
-        %% Stephen Robinsons stuff? this did not work!
-        L2_inv = inv(L2);
-        Z2 = L2_inv * L1 * L2_inv;
-        [U,S] = svds(Z2,1,0);
-        % find the smallest eigenvalue and eigenvector
-        Y = L2_inv*U;
-        Y = Y./sqrt(dot(Y,Y));
-        U = Y;
-
+        % Use Sekihara's method of finding the optimum orientation
+        %
+        % Sekihara et al. Asymptotic SNR of scalar and vector minimum-variance
+        % beamformers for neuromagnetic source reconstruction. IEEE Trans. Biomed.
+        % Eng, No 10, Vol. 51, 2004, pp 1726-1734
+        [U,S] = svd(pinv(lf' * inv_cov * lf)); % JM added pinv, in order to take the orientation that belong to the maximum S
+         
+      case'moiseev'
+        if ~isempty(toi)
+          % use trial averaged variance matrix within a time of interest
+          Avg   = dat(:,toi(1):toi(2));
+          Cavg  = Avg*Avg'/size(Avg,2);
+          Sproj = inv_cov * Cavg * inv_cov';
+        else
+          % in case of induced data, just use the Signal cov only
+          Sproj = inv_cov;
+        end
+        
+        Nproj = inv_cov * noisecov * inv_cov';
+        
+        Y1 = lf' * Sproj * lf;
+        Y2 = lf' * Nproj * lf;
+        % find the eigenvalues and eigenvectors
+        [U,S] = eig(Y1,Y2);
+        
       otherwise
-        ft_error('unknown orimethod');
+        ft_error(sprintf('unknown fixedori method %s', fixedori));
     end
-
-    % The optimum orientation is the eigenvector that corresponds to the
-    % smallest eigenvalue.
-
-    % Double check that this is the case, because for single sphere head
-    % model, one of the eigenvectors corresponds to the radial direction,
-    % giving lead fields that are zero (to within machine precission).
-    % The eigenvalue corresponding to this eigenvector can actually be
-    % the smallest and can give the optimum (but wrong) Z-value!)
-
-    ori1 = U(:,1); ori1 = ori1/norm(ori1);
-    ori2 = U(:,2); ori2 = ori2/norm(ori2);
-    % ori3 = U(:,3); ori3 = ori3/norm(ori3);
-
-    L1 = L * ori1;
-    L2 = L * ori2;
-    % L3 = L * ori3;
-
-    if (norm(L1)/norm(L2)) < 1e-6
-      % the first orientation seems to be the silent orientation
-      % use the second orientation instead
-      opt_vox_or = ori2;
-    else
-      opt_vox_or = ori1;
+      
+    % convert the U matrix into the optimal orientation vector
+    switch fixedori
+      case 'spinning'
+        % do nothing, optimum orientation is already computed above
+      otherwise
+        
+        % The optimum orientation is the eigenvector that corresponds to the
+        % biggest eigenvalue (biggest value is more logical, as it relates to SNR).
+        
+        if isfield(headmodel,'singlesphere') && isempty(noisecov) && ~strcmp(fixedori,'moiseev')
```

Comment:
&& isempty(noisecov) conditional should be removed, it will always be false at it is created above

Reply:
OK, I removed it.